### PR TITLE
Leases: transfer tokens

### LINKED ISSUE #235: leases: tokens not transferring
sad-trombone.wav

### PR DESCRIPTION
**Pull request by boz**
fixes #235

### CODE REVIEW

File: app/app_test.go [hunk 1]
```diff
@@ -89,3 +95,80 @@ func TestApp(t *testing.T) {
...
+
+	_, keyfrom := testutil.PrivateKeySigner(t)
+	addrfrom := keyfrom.PubKey().Address().Bytes()
+
+	keyto := testutil.PrivateKey(t)
+	addrto := keyto.PubKey().Address().Bytes()
+
+	commitState, cacheState := testutil.NewState(t, &types.Genesis{
+		Accounts: []types.Account{
+			types.Account{Address: addrfrom, Balance: balance, Nonce: nonce},
```
Comment (golangci) [inline app/app_test.go]: File is not gofmt-ed with -s

File: app/app.go [hunk 2]
```diff
@@ -241,12 +252,15 @@ func (app *app) EndBlock(req tmtypes.RequestEndBlock) tmtypes.ResponseEndBlock {
 func (app *app) Commit() tmtypes.ResponseCommit {
 	app.trace("Commit")
 
-	err := app.cacheState.Write()
-	if err != nil {
-		app.log.Error("error when writing to cache")
+	if err := lease.ProcessLeases(app.cacheState); err != nil {
```
Comment (aastein) [inline app/app.go]: why use cacheState here?

Comment (boz) [reply app/app.go]: thought was to keep all modifications in the ephemeral db until commit.  should we be using `commitState`?

Comment (aastein) [reply app/app.go]: I thought the process lease routine sends transactions, but it modifies state directly. Using cache state is the correct thing to do 👍

**Pull request merged**